### ISSUE TITLE
Great plan and design a bitcoin wallet

### ISSUE THREAD

**Issue by DATSEC** [bitcoin/bitcoin#9948]
I have a very interesting idea for the future, which should be implemented. 
I propose to partially rewrite the wallet to another, the language used by browsers. 
Why this is important. 
You all know that web sites are evolving such languages as html, css, javascript will be better and better. 
If we rewrite the bitcoin wallet on this language, we can create a truly beautiful design. 
There will be charts and indicators, and the load on the network wallet will be like a giant site with indicators. 

If the idea of p2pool is implemented , there will be more indicators of the pool and the capacity of the network. 
Is that money going long because of the load on the network, you can specify its transaction priority, and the amount. 
Then miners will switch automatically to these addresses may take a long time and will go to a bonus from a user specified priority. 
What it will give. 

Sometimes people need money urgently and the translation should be as quick, and as he is willing to donate some money, this priority will allow you to process transactions with maximum speed, allowing the miner to earn more. 

That you would understand what I mean, look at the editor https://atom.io/ he's got the same design on almost all operating systems. 
There is also the function of the plugins, very cool idea and bitcoin, if you need the indicator, put the plugin, if you need p2pool inside of the wallet, put the plug-in. 

Cats will not be crammed with all and Sundry that nobody needs, each user decides what he needs and install what he wants.

Sory for translation.

**Comment by laanwj** [comment]
Can you please stop opening issues for blue-sky ideas like this? This is not what the issue tracker is for, it is primarily to track bugs and critical problems.

**Comment by DATSEC** [comment]
@laanwj I thought here you can suggest ideas, isn't it?
if not, I won't offer more.

**Comment by MarcoFalke** [comment]
@DATSEC You are mixing up different ideas. p2pool, fees and the gui.
Feel free to implement new features, but this issue tracker is mostly
for bugs or critically needed features.

Also note that there are pulls open to rework the gui, such as #8550

**Comment by laanwj** [comment]
> @laanwj I thought here you can suggest ideas, isn't it? if not, I won't offer more.

Within limits, yes.
Something like "I'd like a button to show bitcoin.conf" is a reasonable feature request.
But for very big things that combine things all over the place like this it makes no sense, unless you're planning to implement it yourself and asking for input.

**Comment by DATSEC** [comment]
But the idea is cool or not?
I'm not asking to do it now, just a thought described how it would be possible to do in the future.

**Comment by sipa** [comment]
There are many things that are cool. They're not all feasible, certainly not if you don't need intend to program them youself.

**Comment by DATSEC** [comment]
Maybe soon I can help

**Comment by sipa** [comment]
You're producing vague ideas. Am I correctly summarizing your idea as 1) P2Pool inside the wallet would be cool 2) A wallet that looks more like a website would be cool 3) Using atom.io would be cool?

That's not how it works.

How will P2Pool be integrated in the wallet? Are you going to rewrite it from scratch? Who will do that work (p2pool took months if not more to develop)? Who will use it, given that hardware owners these days don't even run bitcoin nodes anymore?

Writing the wallet in a web-like way... ok... Qt (what we're using now for the GUI) allows styling using css and other web-like elements already. What do you suggest doing? Rewrite it? Tweak it? Do you have any plan for accomplishing that?

In short: vague ideas are easy. The hard part is executing them. Show where you can help by actually improving things, not by randomly shouting words that sound cool together.

**Comment by DATSEC** [comment]
You can use Python
https://github.com/IronLanguages/main

**Comment by sipa** [comment]
Stop spamming.

**Comment by DATSEC** [comment]
@sipa So what's the problem to use Python for such purposes?

**Comment by sipa** [comment]
@DATSEC Nothing. But you need to elaborate. "Hey we can use Python for the wallet!" is similar to saying "Hey we can use wood for buildings!". Yes, of course you can.

Stop saying random words and contribute.

**Comment by DATSEC** [comment]
@sipa The idea is not to achieve the goal and in its journey.
In the end, it's only a matter of time.
